spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/provisioning/PubSubChannelProvisioner.java

```diff
@@ -54,8 +58,12 @@ public ProducerDestination provisionProducerDestination(String name,
 	public ConsumerDestination provisionConsumerDestination(String name, String group,
 			ExtendedConsumerProperties<PubSubConsumerProperties> properties)
 			throws ProvisioningException {
+		// Use group name as subscription name
+		// Generate anonymous random group, if one not provided
+		String subscription = !StringUtils.hasText(group) ?
+				"anonymous." + name + "." + UUID.randomUUID().toString()
```

Comment:
Why exactly not start from the topic name first and only then an anonymous suffix?

Just wonder if that would better: I can live with this as well!

Reply:
I thought that if a topic name itself ends with ".anonymous", this can cause ambiguity for parsing, if one ever needs to do it.

Reply:
OK. Makes sense.

Closing conversation...